silkworm/rpc/core/evm_trace.cpp

```diff
@@ -778,6 +780,23 @@ void VmTraceTracer::on_execution_end(const evmc_result& result, const silkworm::
             }
             break;
     }
+
+    /* EVM WA: EVMONE add OP_STOP at the end of tx if not present but doesn't notify to the tracer. Add sw to add STOP to the op list */
+    if (result.status_code == EVMC_SUCCESS && last_opcode_ && last_opcode_ != OP_SELFDESTRUCT && last_opcode_ != OP_RETURN && last_opcode_ != OP_STOP) {
```

Comment:
I suggest extracting this condition in a dedicated function not to repeat it twice. Next PR is fine